pkg/controller/container-runtime-config/container_runtime_config_controller_test.go

```diff
@@ -830,30 +832,88 @@ func TestRegistriesValidation(t *testing.T) {
+						},
+					},
+				},
+			},
+		},
 	}
 
 	successTests := []struct {
-		name   string
-		config *apicfgv1.RegistrySources
+		name                                                              string
+		expectedRegistriesBlocked, expectedPolicyBlocked, expectedAllowed []string
+		config                                                            *apicfgv1.RegistrySources
+		icspRules                                                         []*apioperatorsv1alpha1.ImageContentSourcePolicy
 	}{
 		{
-			name: "adding registry used by payload to blocked registries",
+			name: "adding registry used by payload to insecure registries",
 			config: &apicfgv1.RegistrySources{
 				BlockedRegistries:  []string{"docker.io"},
 				InsecureRegistries: []string{"blah.io"},
 			},
+			expectedRegistriesBlocked: []string{"docker.io"},
+			expectedPolicyBlocked:     []string{"docker.io"},
+		},
+		{
+			name: "adding registry used by payload to blocked registries with mirror rules configured for the payload reg",
+			config: &apicfgv1.RegistrySources{
+				BlockedRegistries:  []string{"blah.io", "docker.io"},
+				InsecureRegistries: []string{"test.io"},
+			},
+			icspRules: []*apioperatorsv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: apioperatorsv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []apioperatorsv1alpha1.RepositoryDigestMirrors{
+							{Source: "blah.io/payload", Mirrors: []string{"mirror-1.io/payload", "mirror-2.io/payload"}},
+						},
+					},
+				},
+			},
+			expectedRegistriesBlocked: []string{"blah.io", "docker.io"},
+			expectedPolicyBlocked:     []string{"blah.io", "docker.io"},
+			expectedAllowed:           []string{"blah.io/payload/myimage"},
+		},
+		{
+			name: "adding payload repository to blocked registries with mirror rules configured for the payload registry",
+			config: &apicfgv1.RegistrySources{
+				BlockedRegistries:  []string{"blah.io/payload", "docker.io"},
+				InsecureRegistries: []string{"test.io"},
+			},
+			icspRules: []*apioperatorsv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: apioperatorsv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []apioperatorsv1alpha1.RepositoryDigestMirrors{
+							{Source: "blah.io", Mirrors: []string{"mirror-1.io", "mirror-2.io"}},
+						},
+					},
+				},
+			},
+			expectedRegistriesBlocked: []string{"blah.io/payload", "docker.io"},
+			expectedPolicyBlocked:     []string{"blah.io/payload", "docker.io"},
+			expectedAllowed:           []string{"blah.io/payload/myimage"},
 		},
 	}
 
 	// Failure Tests
 	for _, test := range failureTests {
 		imgcfg := newImageConfig(test.name, test.config)
-		cvcfg := newClusterVersionConfig("version", "blah.io/myuser/myimage:test")
-		blocked, err := getValidBlockedRegistries(cvcfg.Status.Desired.Image, &imgcfg.Spec)
+		cvcfg := newClusterVersionConfig("version", "blah.io/payload/myimage:test")
```

Comment:
These tests (really, all callers of `newClusterVersionConfig`) should now use digested references to be representative; otherwise `PullSourcesFromReference`’s tagged/digested configuration would interpret the configuration in ways that don’t match production.

Reply:
fixed